research/src/pages/radio-button/radio-button.proposal.mdx

```diff
@@ -0,0 +1,185 @@
+---
+menu: Proposals
+name: Radio Button (Editor's Draft)
+path: /components/radio-button
+pathToResearch: /components/radio-button.research
+---
+
+import RadioButtonAnatomy from '../../components/radio-button-anatomy'
+import RadioButtonGroupAnatomy from '../../components/radio-button-group-anatomy'
+
+## Overview
+
+The `<oui-radio-button>` element is a control that allows the user to select a single choice out of a set of choices. It has a binary "checked" state, either checked or unchecked (where "checked" is the selected choice, while all the other remain "unchecked".
+
+Because the `<oui-radio-button>` is a choice from a set of choices, the choices are logically grouped by an implicit or explicit `<oui-radio-button-group>`. This proposal will deal with the existence of both elements, the different ways they can interact and their properties.
+
+### Use Cases
+
+The `<oui-radio-button-group>` is used in cases very similar to a `<select>` element, where the user is presented with a set of options from which they can select a single one.
+
+Differently from `<select>`, where all the `<option>` are condensed into a single box, the `<oui-radio-button-group>` has all the `<oui-radio-button>` visible, readily available to be read and selected. Choosing between a `<select>` and `<oui-radio-button-group>` is a matter of layout and information density preference, in most cases.
+
+Also differently from a `<select>`, there is currently no `multiple` attribute to allow a multiple choice group, and this document will not propose one for now.
+
+## Prior Art/Examples
+
+- [Radio input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio)
+- [WAI-ARIA Radio](https://www.w3.org/TR/wai-aria-practices/#radiobutton)
+- [WHATWG](<https://html.spec.whatwg.org/multipage/input.html#radio-button-state-(type=radio)>)
+
+[Further examples on research](/components/radio-button.research).
+
+## Radio Button Design
+
+This section relates to a single `<oui-radio-button>`. Because it cannot stand on its own, without being part of and either explicit or implicit `<oui-radio-button-group>`, it does not have attributes such `required` or `readonly`, those being attributes of the group. A single `required` option is a group makes the other choices pointless, while a `readonly` option makes itself pointles.
+
+The radio button has, however a `disabled` attribute, meaning it's a possible choice, but that is currently unavailable.
+
+If used inside a `<oui-radio-button-group>`, it gets implicitly bound to the group, becoming one of the choices. If declared outside of it, it should contain a `group` attribute equal to the group's `name` attribute.
```

Comment:
Hello @scottaohara , how are you?

During conversations about this proposal, a topic came up, in which the `name` attribute might be reserved for other purposes, or at least should be re-thought.

Also, because the proposal changes a lot of other aspects, it is expected that it's not exactly like the current standard.

Reply:
Understood, as I did see the mention of the attribute `group`.  My mention of `name`'s current behavior was in regards to the fact that this thread mentioned uncertainty in how to associated radios if not children of a `radiogroup`.  Being that `name` makes this possible now seemed appropriate to bring up since there seemed uncertainty.

Per this, I've also filed an issue against ARIA https://github.com/w3c/aria/issues/1721 to allow for a feature with ARIA to make element associations without the need for explicit nesting within a parent group.

Reply:
Oh, I might not have been clear, but I understood `name`'s current behavior :)

> Per this, I've also filed an issue against ARIA https://github.com/w3c/aria/issues/1721 to allow for a feature with ARIA to make element associations without the need for explicit nesting within a parent group.

This would be exactly what we need to make it ARIA compliant, thanks!